common/meta/api/src/schema_api_impl.rs

```diff
@@ -1494,6 +1563,61 @@ impl<KV: KVApi> SchemaApi for KV {
         }
     }
 
+    /// Get the count of tables for one tenant.
+    ///
+    /// Accept tenant name and returns the count of tables for the tenant.
+    ///
+    /// It get the count from kv space first,
+    /// if not found, it will compute the count by listing all databases and table ids.
+    async fn count_tables(&self, req: CountTablesReq) -> Result<CountTablesReply, MetaError> {
+        let key = CountTablesKey {
+            tenant: req.tenant.to_string(),
+        };
+
+        let count = loop {
+            let (seq, cnt) = {
+                // get the count from kv space first
+                let (seq, c) = get_u64_value(self, &key).await?;
+                if seq > 0 {
+                    // if seq > 0, we can get the count directly
+                    break c;
+                }
+
+                // if not, we should compute the count from by listing all databases and table ids
+
+                // this line of codes will only be executed once,
+                // because if `send_txn` failed, it means another txn will put the count value into the kv space，
+                // and then the next loop will get the count value through `get_u64_value`.
+                (0, count_tables(self, &key).await?)
+            };
+
+            let key = CountTablesKey {
+                tenant: req.tenant.clone(),
+            };
+
+            let txn_req = TxnRequest {
```

Comment:
why `count_tables` api need txn?IMHO, `count_tables` is only a read operation.

Reply:
The increment of table counts is done with other write operations in a transaction. So I think we should read in a transcation. If we don't need strong timeliness about table counts, we can read without a transaction?

Reply:
How do you think about this @drmingdrmer ?